models/maespa/R/met2model.MAESPA.R

```diff
@@ -29,172 +29,180 @@
-
-}  ### End of function
+  
+ print("internal results")
+ print(results)
+                             
+ if (file.exists(out.file.full) && !overwrite) {
+                               logger.debug("File '", out.file.full,
+                                            "' already exists, skipping to next file.")
+                               return(invisible(results))
+                             }
+                             
+                             
+ library(ncdf4)
+ library(lubridate)
+ library(PEcAn.data.atmosphere)
+ library(Maeswrap)
+ 
+ ## check to see if the outfolder is defined, if not create directory for output
+ if (!file.exists(outfolder)) {
+   dir.create(outfolder)
+ }
+ 
+ out <- NULL
+ 
+ # get start/end year since inputs are specified on year basis
+ start_year <- year(start.date)
+ end_year <- year(end.date)
+ 
+ ## loop over files
+ for (year in start_year:end_year) {
+   print(year)
+   
+   old.file <-
+     file.path(in.path, paste(in.prefix, year, "nc", sep = "."))
+   
+   if (file.exists(old.file)) {
+     ## open netcdf
+     nc <- nc_open(old.file)
+     ## convert time to seconds
+     sec   <- nc$dim$time$vals
+     sec <- udunits2::ud.convert(sec,unlist(strsplit(nc$dim$time$units," "))[1],"seconds")
+     
+     ifelse(
+       leap_year(year),
+       dt <-(366 * 24 * 60 * 60) / length(sec), #leap year
+       dt <-(365 * 24 * 60 * 60) / length(sec)
+     ) #non-leap year
+     tstep <- round(86400 / dt)
+     dt <- 86400 / tstep
+     
+     #Check wich variabales are available and which are not
+     
+     ## extract variables
+     lat <- ncvar_get(nc,"latitude")
+     lon <- ncvar_get(nc,"longitude")
+     RAD <- ncvar_get(nc,"surface_downwelling_shortwave_flux_in_air") #W m-2
+     PAR <- try(ncvar_get(nc,"surface_downwelling_photosynthetic_photon_flux_in_air")) #mol m-2 s-1
+     TAIR <- ncvar_get(nc,"air_temperature") # K
+     QAIR <- ncvar_get(nc,"specific_humidity")# 1
+     PPT <- ncvar_get(nc,"precipitation_flux") #kg m-2 s-1
+     CA <- try(ncvar_get(nc,"mole_fraction_of_carbon_dioxide_in_air")) #mol/mol
+     PRESS <- ncvar_get(nc,"air_pressure")# Pa
+     
+     ##Convert specific humidity to fractional relative humidity
+     RH <- qair2rh(QAIR,TAIR,PRESS)
+ 
+     ## Process PAR
+     if (!is.numeric(PAR)) {
+       #Function from data.atmosphere will convert SW to par in W/m2
+       PAR <- sw2par(RAD)
+     }else{
+       #convert
+       PAR <- udunits2::ud.convert(PAR,"mol","umol")
+     }
+
+     # Convert air temperature to Celsius 
+     TAIR <- udunits2::ud.convert(TAIR,"kelvin","celsius")
+     
+     ####ppm. atmospheric CO2 concentration. Constant from Enviiron namelist used instead
+     if (!is.numeric(CA)) {
+       print(
+         "Atmospheric CO2 concentration will be set to constant value set in ENVIRON namelist "
+       )
+       rm(CA)
+       defaultCO2 = 400 #400 is estimation of atmospheric CO2 in ppm)
+     } else {
+       defaultCO2 = 400
```

Comment:
I'm confused here and line 198 -- it appears you're using the default CO2 value even if CO2 data is provided.